subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/SnapshotTaskInputsOperationIntegrationTest.groovy

```diff
@@ -343,6 +363,84 @@ class SnapshotTaskInputsOperationIntegrationTest extends AbstractIntegrationSpec
         }
     }
 
+    def "exposes file inputs, not ignoring empty directories"() {
+        given:
+        withBuildCache()
+        settingsFile << "include 'a'"
+        createDir("a") {
+            file("build.gradle") << """
+                task foo {
+                    inputs.dir('src').ignoreEmptyDirectories(false).withPropertyName('src')
+                    outputs.file('output.txt')
+                    doLast {
+                        file('output.txt') << 'do stuff'
+                    }
+                }
+            """
+            dir("src") {
+                file("A.txt") << "fooA"
+                file("B.txt") << "fooB"
+                dir("a") {
+                    file("A.txt") << "fooA"
+                    dir("a") {
+                        file("A.txt") << "fooA"
+                    }
+                }
+                dir("empty") {
+                    dir("empty")
+                }
+            }
+        }
+
+        when:
+        succeeds("a:foo")
+
+        then:
+        def result = snapshotResults(":a:foo")
+        def aFoo = result.inputFileProperties
+        with(aFoo.src) {
+            hash != null
+            attributes == ['DIRECTORY_SENSITIVITY_DEFAULT', 'FINGERPRINTING_STRATEGY_ABSOLUTE_PATH', 'LINE_ENDING_SENSITIVITY_DEFAULT']
```

Comment:
```suggestion
            attributes == ['DIRECTORY_SENSITIVITY_IGNORE_DIRECTORIES', 'FINGERPRINTING_STRATEGY_ABSOLUTE_PATH', 'LINE_ENDING_SENSITIVITY_DEFAULT']
```

Shouldn't this be `DIRECTORY_SENSITIVITY_IGNORE_DIRECTORIES`?

Reply:
No, it's set like `inputs.dir('src').ignoreEmptyDirectories(false).withPropertyName('src')`

Reply:
Ah, I somehow missed the false. I suppose the Java compile test is then for ignoring empty directories, right?